swap_meet/vendor.py

```diff
@@ -1,2 +1,85 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory or []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        
+        self.inventory.remove(item)
+        return item
+    
+    def get_by_id(self, num=None):
+        for item in self.inventory:
+            if item.id == num:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+        other_vendor.add(my_item)
+        self.add(their_item)
+        other_vendor.remove(their_item)
+        self.remove(my_item)
+        
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+
+        self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+
+        return True
+
+    def get_by_category(self, category):
+
+        items = [item for item in self.inventory if item.get_category() == category]
+        return items
+    
+    def get_best_by_category(self, category):
+
+        items = self.get_by_category(category)
+
+        if not items:
+            return None
+        
+        best_item = None
+        for item in items:
+            if best_item is None or item.condition > best_item.condition:
+                best_item = item
```

Comment:
A super small tweak, but if we can avoid having to check and see if best_item is None along with the more important condition of comparing the conditions, we should try that!